odbc/src/api/get_type_info_tests.rs

```diff
@@ -0,0 +1,182 @@
+use crate::{handles::definitions::*, SQLGetTypeInfo};
+use bson::Bson;
+use odbc_sys::{SqlDataType, SqlReturn};
+
+mod unit {
+    use super::*;
+    use mongo_odbc_core::Error;
+
+    fn validate_result_set(data_type: SqlDataType, expectations: Vec<&str>) {
```

Comment:
You don't need to change this but just for your own understanding of our typical approach to unit testing (and possibly a common Rust test paradigm): you _could_ refactor this helper function into a macro that produces all of the test functions for you! _Again, in this case, you don't actually need to do this change since I think ultimately the result would be the same number of lines of code **and** this is a very readable helper function and the call sites are very readable._ The way you _could_ make this change would be having a macro that generates the `#[test] fn test_name() { ... }` code for you, where the `...` is effectively wholesale replaced with the code you have in this helper function's body. The parameters of the macro would be test_name, data_type, and expectations. See [here](https://github.com/mongodb/mongo-odbc-driver/blob/36e6a5ed3ea0a83579c652d8a46afd44712a668c/odbc/src/api/connect_attr_tests.rs#L15) or [here](https://github.com/10gen/mongosql-rs/blob/927ec68cb9b48978e9ed806635c8c851a7dcb0cf/mongosql/src/translator/test.rs#L1) for examples of this in both our rust codebases.

I think I actually prefer this written this way in this case since it is only used a few times and the tests are all already well name and organized 👍 . Just sharing so you have the macro-model of unit testing in your tool belt for the future.

Reply:
Thanks for laying this all out, this is very helpful. I had used a macro for testing on one of my very first tickets, but was just following an existing pattern. This really helps clarify!